src/test/resources/com/puppycrawl/tools/checkstyle/checks/modifier/redundantmodifier/InputRedundantModifierNestedDef.java

```diff
@@ -15,4 +15,16 @@
     interface MyInnerInterface {
         public strictfp class MyInnerClass {} // violation
     }
+
+    public static class testClass { // 2 violations
+    }
+
+    public abstract static @interface testAnnotatedInterface { // 3 violations
```

Comment:
what about cases like
```
interface A { // top level
   interface B {
       static interface C {} // will be violation?
   }
}
```

Reply:
> will be violation?

Yes.

The level of nesting doesn't matter, as long as it is nested inside an interface, it is public and static implicitly.

Reply:
please extend test inputs to show this

Reply:
Done